sdk/src/lib.rs

```diff
@@ -85,68 +86,92 @@ impl ProverClient {
     pub fn prove(&self, elf: &[u8], stdin: SP1Stdin) -> Result<SP1ProofWithIO<BabyBearPoseidon2>> {
         if self.client.is_some() {
             println!("Proving remotely");
-            self.prove_remote(elf, stdin)
+            self.prove_remote_sync(elf, stdin)
         } else {
             println!("Proving locally");
             self.prove_local(elf, stdin, BabyBearPoseidon2::new())
         }
     }
 
-    pub fn prove_remote(
+    /// Generate a proof for the execution of the ELF with the given public inputs for an async context
+    pub async fn prove_async(
+        &self,
+        elf: &[u8],
+        stdin: SP1Stdin,
+    ) -> Result<SP1ProofWithIO<BabyBearPoseidon2>> {
+        if self.client.is_some() {
+            println!("Proving remotely");
+            self.prove_remote(elf, stdin).await
+        } else {
+            println!("Proving locally");
+            self.prove_local(elf, stdin, BabyBearPoseidon2::new())
```

Comment:
I think `prove_async` is is technically inaccurate in this context because `prove_local` is sync, maybe there is a better way to do it?

Reply:
What do you think the interface should be for `prove_async`?

Either:
- Add `prove_local_async`
- Remove `prove_local` from `prove_async` and only expose `prove_remote_async`

Reply:
Removed `prove_async` for now.